packages/core/content-manager/admin/src/pages/EditView/components/FormInputs/BlocksInput/BlocksToolbar.tsx

```diff
@@ -538,32 +667,104 @@ const BlocksToolbar = () => {
 
   const isButtonDisabled = checkButtonDisabled();
 
+  /**
+   * Observed items are ones that may or may not be visible in the toolbar, depending on the
+   * available space. They provide two render props:
+   * - renderInToolbar: for when we try to render the item in the toolbar (may be hidden)
+   * - renderInMenu: for when the item didn't fit in the toolbar and is relegated to the "more" menu
+   */
+  const observedItems = [
+    ...Object.entries(modifiers).map(([name, modifier]) => ({
+      renderInToolbar: () => (
+        <ToolbarButton
+          key={name}
+          name={name}
+          icon={modifier.icon}
+          label={modifier.label}
+          isActive={modifier.checkIsActive(editor)}
+          handleClick={() => modifier.handleToggle(editor)}
+          disabled={isButtonDisabled}
+        />
+      ),
+      renderInMenu: () => (
+        <Menu.Item onClick={() => modifier.handleToggle(editor)}>
+          {formatMessage(modifier.label)}
+        </Menu.Item>
+      ),
+    })),
+    {
+      renderInToolbar: () => <LinkButton disabled={isButtonDisabled} location="toolbar" />,
+      renderInMenu: () => <LinkButton disabled={isButtonDisabled} location="menu" />,
+    },
+    {
+      // List buttons can only be rendered together when in the toolbar
+      renderInToolbar: () => (
```

Comment:
do these need to be functions? Couldn't they just be object keys? That would also match the location prop thing you got going. 

```
toolbar: <ToolbarComponent />
menu: <MenuComponent />
```
Then used like:
```tsx
 {observedItems
              .map((item, index) => (
                <ObservedToolbarItem
                  lastVisibleIndex={lastVisibleIndex}
                  setLastVisibleIndex={setLastVisibleIndex}
                  index={index}
                  rootRef={toolbarRef}
                  key={index}
                >
                  {item['toolbar']} <---------here!
                </ObservedToolbarItem>
              ))
```